### PR TITLE
Adding new file_type functionality

### LINKED ISSUE #4: Add bam input option
Will use bam instead of fastq. Default remains fastq to not break any scripts already implemented.

### LINKED ISSUE #5: Update CLI argument parsing to optparse from base R positional stuff
What it says in the title

### PR DESCRIPTION
**Pull request by aidaanva**
- It adds new functionality to allow for changing the table to contain bam or fastq files processed by the Autorun pipelines [closes #4]
- Includes parsing of arguments [closes #5]

### CODE REVIEW

File: exec/pandora2eager.R [hunk 1]
```diff
@@ -9,33 +9,53 @@ library(readr)
 library(tidyr)
 library(stringr)
 library(pandora2eager)
+library(optparse)
+
+# Function to validate the file type
+validate_file_type <- function(option, opt_str, value, parser) {
+  valid_entries <- c("bam","fastq_pathogens") ## TODO comment: should this be embedded within the function? You would want to maybe update this over time no?
```
Comment (TCLamnidis) [inline exec/pandora2eager.R:16]: Indeed, the valid entries could be passed in as an argument to the function. Feel free to make the change if you want.

**Review by TCLamnidis — COMMENTED** [review]
LGTM. Will test the container version and make any changes needed before merging.

File: exec/pandora2eager.R (hunk 1 above)
Comment (aidaanva) [inline exec/pandora2eager.R:16]:
```suggestion
  valid_entries <- c("bam","fastq_pathogens")
```

Comment (aidaanva) [reply exec/pandora2eager.R]: I think since it does not really matter, I will leave it within the function

File: exec/pandora2eager.R [hunk 2]
```diff
@@ -48,48 +68,88 @@ collect_and_format_info<- function(query_list_seq, con) {
...
+      filter(!grepl("_Libmerge_Genotypes",analysis.Result_Directory)) %>%
+      select(seqID, analysis.Result_Directory) %>%
+      ##Remove duplicated lines 
+      distinct()
+
+      results <- results %>%
+      inner_join(analysis_tab, by=c("sequencing.Full_Sequencing_Id"="seqID")
+      ) %>%
+      mutate(Lane=row_number(), 
+      R1="NA",
```
Comment (aidaanva) [inline exec/pandora2eager.R:86]:
```suggestion
      R1=NA, 
```

File: exec/pandora2eager.R [hunk 3]
```diff
@@ -48,48 +68,88 @@ collect_and_format_info<- function(query_list_seq, con) {
...
+      select(seqID, analysis.Result_Directory) %>%
+      ##Remove duplicated lines 
+      distinct()
+
+      results <- results %>%
+      inner_join(analysis_tab, by=c("sequencing.Full_Sequencing_Id"="seqID")
+      ) %>%
+      mutate(Lane=row_number(), 
+      R1=NA, 
+      R2="NA",
```
Comment (aidaanva) [inline exec/pandora2eager.R:87]:
```suggestion
      R2=NA, 
```

File: exec/pandora2eager.R [hunk 4]
```diff
@@ -48,48 +68,88 @@ collect_and_format_info<- function(query_list_seq, con) {
...
+      BAM=paste0(analysis.Result_Directory,sequencing.Full_Sequencing_Id,".bam"), 
+      SeqType="SE")
+    } else if(file=="fastq_pathogens"){
+      analysis_tab <- get_analysis_tab(query_list_seq, con) %>%
+      ##Filter all the analysis.Title that have a fastq with mapped reads (only existing in the prescreening pipelines)
+      filter(analysis.Title=="Fastq mapped reads")
+
+      results <- results %>%
+      inner_join(analysis_tab, by=c("sequencing.Full_Sequencing_Id"="seqID")) %>%
+      mutate(Lane=row_number(), R1= analysis.Result, R2="NA", BAM="NA", SeqType="SE")
```
Comment (aidaanva) [inline exec/pandora2eager.R:98]:
```suggestion
      mutate(Lane=row_number(), R1= analysis.Result, R2=NA, BAM=NA, SeqType="SE")
```

**Pull request merged**